Sources/SwiftDocC/Infrastructure/Symbol Graph/GeneratedDocumentationTopics.swift

```diff
@@ -225,12 +246,11 @@ enum GeneratedDocumentationTopics {
                let child = context.symbolIndex[relationship.source],
                // Get the swift extension data
                let extends = child.symbol?.mixins[SymbolGraph.Symbol.Swift.Extension.mixinKey] as? SymbolGraph.Symbol.Swift.Extension {
-                var originParentSymbol: ResolvedTopicReference? = nil
-                if let originSymbol = context.symbolIndex[origin.identifier] {
-                    originParentSymbol = parentOfFunction(originSymbol.reference)
-                }
+                let originSymbol = context.symbolIndex[origin.identifier]?.symbol
+                let sourceSymbol = context.symbolIndex[relationship.source]?.symbol
```

Comment:
This is checked above as `let child = context.symbolIndex[relationship.source]` so it should never be `nil`. 

I think this means that the `sourceSymbol` argument to `add(...)` could also be non-optional which removes the need for the `else` case in that implementation.

Reply:
Oh nice catch! Thanks. Addressed here: https://github.com/apple/swift-docc/pull/321/commits/976c858636a83050ebfcb879fbed58a97b0e1557.

I think we should still leave the `else` to reduce risk here in case the child symbol is an unexpected length but this definitely make things clearer.